Source/Core/Common/BitUtils.h

```diff
@@ -154,14 +154,27 @@ class BitCastPtrType
     return result;
   }
 
+  inline auto operator[](std::size_t index) const
```

Comment:
`inline` is redundant here, though I realize you are probably just copying the pattern of the other functions.

This whole template could really use some modernization, but that can be saved for another PR.